bin/ebuild.sh

```diff
@@ -50,6 +51,19 @@ __check_bash_version() {
 	if ___eapi_bash_3_2 && [[ ${BASH_VERSINFO[0]} -gt 3 ]] ; then
 		shopt -s compat32
 	fi
+
+	# patsub_replacement is a new option in bash-5.2 that is also default-on
+	# in that release. The default value is not gated by BASH_COMPAT (see bug #881383),
+	# hence we need to disable it for older Bashes to avoid behaviour changes in ebuilds
+	# and eclasses.
+	#
+	# New EAPI note: a newer EAPI (after 8) may well adopt Bash 5.2 as its nimimum version.
+	# If it does, this logic will need to be adjusted to only disable patsub_replacement
+	# for < ${new_api}!
+	if [[ ${BASH_VERSINFO[0]} -ge 6 ]] || [[ ${BASH_VERSINFO[0]} -eq 5 && ${BASH_VERSINFO[1]} -eq 2 ]] ; then
```

Comment:
I don't think we can/want to, because then we have to check for older EAPIs and I don't really want to go verify that none of those functions then call us. But it was a good idea.

Reply:
Right, that does make sense. I didn't think of that, sorry.